cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1562 @@
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/common/crypto"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
+	"github.com/thrasher-corp/gocryptotrader/log"
+	"golang.org/x/net/html"
+)
+
+const (
+	githubPath         = "https://api.github.com/repos/%s/commits/master"
+	jsonFile           = "updates.json"
+	testJSONFile       = "testupdates.json"
+	backupFile         = "backup.json"
+	github             = "GitHub Sha Check"
+	htmlScrape         = "HTML String Check"
+	pathOkCoin         = "https://www.okcoin.com/docs/en/#change-change"
+	pathOkex           = "https://www.okex.com/docs/en/#change-change"
+	pathBTSE           = "https://www.btse.com/apiexplorer/spot/#btse-spot-api"
+	pathBitfinex       = "https://docs.bitfinex.com/docs/changelog"
+	pathBitmex         = "https://www.bitmex.com/static/md/en-US/apiChangelog"
+	pathANX            = "https://anxv3.docs.apiary.io/"
+	pathPoloniex       = "https://docs.poloniex.com/#changelog"
+	pathIbBit          = "https://api.itbit.com/docs"
+	pathBTCMarkets     = "https://api.btcmarkets.net/openapi/info/index.yaml"
+	pathEXMO           = "https://exmo.com/en/api/"
+	pathBitstamp       = "https://www.bitstamp.net/api/"
+	pathHitBTC         = "https://api.hitbtc.com/"
+	pathBitflyer       = "https://lightning.bitflyer.com/docs?lang=en"
+	pathLakeBTC        = "https://www.lakebtc.com/s/api_v2"
+	pathKraken         = "https://www.kraken.com/features/api"
+	pathAlphaPoint     = "https://alphapoint.github.io/slate/#introduction"
+	pathYobit          = "https://www.yobit.net/en/api/"
+	pathLocalBitcoins  = "https://localbitcoins.com/api-docs/"
+	pathGetAllLists    = "https://api.trello.com/1/boards/%s/lists?cards=none&card_fields=all&filter=open&fields=all&key=%s&token=%s"
+	pathNewCard        = "https://api.trello.com/1/cards?idList=%s&name=%s&key=%s&token=%s"
+	pathChecklists     = "https://api.trello.com/1/checklists/%s/checkItems?%s&key=%s&token=%s"
+	pathChecklistItems = "https://api.trello.com/1/checklists/%s?fields=name&cards=all&card_fields=name&key=%s&token=%s"
+	pathUpdateItems    = "https://api.trello.com/1/cards/%s/checkItem/%s?%s&key=%s&token=%s"
+	pathCheckBoardID   = "https://api.trello.com/1/members/me/boards?key=%s&token=%s"
+	pathNewChecklist   = "https://api.trello.com/1/checklists?idCard=%s&name=%s&key=%s&token=%s"
+	pathNewList        = "https://api.trello.com/1/lists?name=%s&idBoard=%s&key=%s&token=%s"
+	pathGetCards       = "https://api.trello.com/1/lists/%s/cards?key=%s&token=%s"
+	pathGetChecklists  = "https://api.trello.com/1/cards/%s/checklists?&key=%s&token=%s"
+	complete           = "complete"
+	incomplete         = "incomplete"
+	createList         = "UpdatesList"
+	createCard         = "UpdatesCard"
```

Comment:
thoughts on linking these two a config entry and allow the user to override the name via config?